apiserver/application/application.go

```diff
@@ -164,6 +164,22 @@ func (api *API) Deploy(args params.ApplicationsDeploy) (params.ErrorResults, err
 	for i, arg := range args.Applications {
 		err := deployApplication(api.backend, api.stateCharm, arg, api.deployApplicationFunc)
 		result.Results[i].Error = common.ServerError(err)
+
+		if err != nil && len(arg.Resources) != 0 {
+			// Remove any pending resources - these would have been
+			// converted into real resources if the application had
+			// been created successfully, but will otherwise be
+			// leaked. lp:1705730
```

Comment:
add a TODO here to do the application/resource doc creation and removal transactionally, as we talked about?